ambari-server/src/main/java/org/apache/ambari/server/StateRecoveryManager.java

```diff
@@ -39,48 +39,41 @@
   private static final Logger LOG = LoggerFactory.getLogger(StateRecoveryManager.class);
 
   @Inject
-  private HostVersionDAO hostVersionDAO;
+  private MpackHostStateDAO mpackHostStateDAO;
 
+  /**
+   * Used for looking up {@link Mpack} instances by IDs.
+   */
   @Inject
-  private ServiceComponentDesiredStateDAO serviceComponentDAO;
+  private AmbariMetaInfo ambariMetaInfo;
 
   public void doWork() {
-    checkHostAndClusterVersions();
+    updateManagementPackInstallationState();
   }
 
-  void checkHostAndClusterVersions() {
-    List<HostVersionEntity> hostVersions = hostVersionDAO.findAll();
-    for (HostVersionEntity hostVersion : hostVersions) {
-      if (hostVersion.getState().equals(RepositoryVersionState.INSTALLING)) {
-        hostVersion.setState(RepositoryVersionState.INSTALL_FAILED);
+  /**
+   * Resets any management pack installation states from
+   * {@link RepositoryVersionState#INSTALLING} to
+   * {@link RepositoryVersionState#INSTALL_FAILED}.
+   */
+  void updateManagementPackInstallationState() {
+    List<MpackHostStateEntity> mpackHostStates = mpackHostStateDAO.findAll();
+    for (MpackHostStateEntity mpackHostState : mpackHostStates) {
+      if (mpackHostState.getState() == RepositoryVersionState.INSTALLING) {
+        mpackHostState.setState(RepositoryVersionState.INSTALL_FAILED);
```

Comment:
We should consider renaming the enum RepositoryVersionState to some other "thing" - InstallState or something similar.  Doesn't need to be in this change, just bringing it up.  We can use it for multiple things, possibly for mpacks, modules, etc.

Reply:
I agree. I'm trying to keep the number of files changed manageable, so I didn't want to do it as part of this commit.